### ISSUE TITLE
Running an invalid query causes System.exit(0) instead of Exception

### ISSUE THREAD

**Issue by ZsoltBertalan** [realm/realm-kotlin#1339]
### How frequently does the bug occur?
Always

### Description
Running an invalid query causes System.exit(0) instead of Exception

### Stacktrace & log output
```shell
System.exit(0)

That's all you get.
```

### Can you reproduce the bug?
Always

### Reproduction Steps
Run any invalid query

### Version
1.7.0

### What Atlas App Services are you using?
Local Database only

### Are you using encryption?
No

### Platform OS and version(s)
Android 12, API 31

### Build environment
Android Studio version: 2022.2.1 RC1
Android Build Tools version: 30.0.3
Gradle version: 7.5.1

**Comment by rorbech** [comment]
@ZsoltBertalan Could you elaborate on the exact query. When trying to pass in an invalid query expression we throw an exception like 
```
java.lang.IllegalArgumentException: [RLM_ERR_INVALID_QUERY_STRING]: Invalid predicate: 'stringField > $$$ 'Hello'': syntax error, unexpected string, expecting end of file
```
If it crashes there is most probably something else wrong or you hit some corner case in the query or model definition. Could you please provide the invalid query causing the crash and the stack trace from logcat?

**Comment by ZsoltBertalan** [comment]
Indeed it was a very specific mistake on my part. So specific, that I cannot reproduce it anyhow. It happened when I tried to query on backlinks for the first time, which is hard anyway, due to the very limited explanations in the docs. 
I went through my local history, but now everything produces Exceptions. Maybe I remember wrong or there was another issue, of which I cannot find a trace of. I think we can close this for now.
